### ISSUE TITLE
Questions and clarifications for your amazing work.

### ISSUE THREAD

**Issue by dyhan316** [wjq-learning/CBraMod#3]
**Hi, first thank you for the amazing work!** It's quite amazing that how you achieved higher scores compared to LaBraM using much less number of parameters! I am also very glad that you employed rigorous ablation studies, and included the performance when models were pretrained on the same dataset for fair comparison.

We are thinking of using your model on massive amounts of private iEEG data, and had a few **clarification questions** if you don't mind : 

**1. How should I replicate the supervised models?** : It seems you used BIOT to replicate the supervised models (e.g., SPaRCNet, ContraWR, ...). If we want to replicate those results, how should we go about it? It seems that simply using the BIOT code would bring the following challenges. 1. BIOT doesn't include many datasets that you use 2. You also added EEGNet and EEGConformer, which is not included in the BIOT codebase. **Do you by any chance have the code to replicate the supervised model's (and BIOT and LaBraM-Base) results? It would be greatly appreciated!** 

**2. Why is the memory for the full attention version of CBraMod only 0.1M more than that of CBraMod with criss-cross attention?** : In Table 22, it says that the full attention version of the CBraMod has 4.1M parameters, but this is also the same for the criss-cross attention (4.0M). This seems a bit weird. Given that the sequence length is 30, at least for attention the memory ratio should be around 30*30/((30+19)/2) = 36.7 times. It seems likely that I am misunderstanding something?

**3. How does ACPE work when there are less number of channels than what was presented during pretraining?** : I may not understand it fully, but it seems that since your spatial kernel size is $k_s$, if the number of channels is less than $k_s$, you could have trouble. (Or is it OK since you do zero paddings in the channel (spatial) dimension?)

**4. How do you define "neighborhood" channels?** : In your paper you say that "... generate ACPE from the spatial-temporal neighborhoods of an EEG patch, ....". Although I can understand what temporal neighborhood would be, it's hard to grasp what the "spatial" neighborhood would mean. Are they closeby channels? By what measure? If there are more than 19 channels, how does one decide what channels to do depth-wise two-dimensional convolution on? (i.e. how to define the "neighborhood channels" to be put fed into the kernel?)

**5. Mismatch in results SEED-V of LaBraM** : I may be mistaken, but in Table 6 of LaBraM, they state that they achieve Cohen's Kappa value of 0.2613 using LaBraM large for SEED-V. However, your results in Table 2 say that the same model (LaBraM) achieved 0.2386, which is much worse. I am perplexed as to how the performance on the same metric using the same model can differ too much. Am I missing something? 

Below is a not-so-important but **curious question**, please feel free to ignore the questions below (I am just really interested in your model, so wanted to know your "personal" journey during your work and your opinion!) : 
1.  It seems that during pretraining, you did not use the reference electrodes (i.e. A1, A2, or Cz). Wouldn't including them help the model unify the different references together? Have you tested feeding these reference electrode data in?
2. Do you think that better QA/QC control than the simple QA/QC you did would benefit training, or do you think that it wouldn't matter that much?
3. It is interesting that you beat LaBraM while not using tokenizers. What are your opinions on tokenizers for EEG?
4. Have you tried longer sequence lengths? It seems that Table 16 shows longer sequence lengths benefit the model. Why not extend more? Why not try 60s? Was there a reason you didn't try longer sequences? 
5. Is it possible that the length of each patch (1 second) is too short to apply reliable FFT? 
6. Have you tried taking the imaginary FFT too during patch encoding? 
7. Why not concat the temporal and frequency embeddings, $e_{i,j}^t$ and $e_{i,j}^f$ in equation (3)? 
8. It seems that part of your code reference torcheeg but you didn't actually use them. We are thinking of applying your model to iEEG, and was wondering if unifying iEEG datasets into torcheeg should be done. What are you opinins? Do you think torcheeg is worth it?

Sorry for asking a LOT of questions, but **I REALLY enjoyed your papers**, and really am happy you did so many ablation tests and such. **Thank you so much for your work!!**

**Comment by wjq-learning** [comment]
**Hi, first of all, thank you so much for your kind words!** It's truly amazing to hear that you found our work useful and insightful! I’m really glad you raised so many thoughtful questions—some of them are even more detailed than what we get from ICLR reviewers 😄! I’m currently quite tied up with some academic commitments, so I apologize for not being able to respond to all your questions right away. However, I’ll address some of the most pressing ones for now and get back to the rest once I have a bit more time.

---
**1. How should I replicate the supervised models?** :
I apologize that we currently do not have plans to integrate the code for replicating the supervised models (e.g., SPaRCNet, ContraWR, etc.) into our project due to limited resources. However, both BIOT and LaBraM have open-sourced their codebases, and their models are relatively plug-and-play. I believe integrating their models into your workflow should not be overly challenging. For EEGNet and EEGConformer, you may need to implement them separately or adapt existing implementations.

---
**5. Mismatch in results SEED-V of LaBraM**：
This discrepancy is actually addressed in our paper: "_We used the open-source version of SEED-V with 16 subjects, rather than the private version with 20 subjects used in LaBraM_." Please note that the authors of LaBraM are also the owners of the SEED-V dataset, so they had access to a private version with 20 subjects, while we only had access to the publicly available version with 16 subjects. As a result, we had to re-implement all methods on the open-source version, which explains the difference in performance.

---
Thank you again for your thoughtful engagement with our work! I truly appreciate the depth of your questions, and once my academic commitments ease up, I’ll get back to you with more detailed answers to the remaining questions. Thanks again for your interest and for supporting our work!

**Comment by dyhan316** [comment]
Thank you for addressing a few of the important questions, and I look forward your additional responses!

We'll try replicating the other supervised models using the codebase in BIOT but using the dataset and data-loader in your codebase :) 

Also I didn't know that the authors of SEED are authors of LaBraM, how interesting :) Thank you for clearing the SEED-V results thing!
